### PR TITLE
Changed DataTable to add sort and onSort

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed DataTable to add sort and onSort

#### Where should the reviewer start?

DataTable/doc.js

#### What testing has been done on this PR?

Added DataTable Sort story

#### How should this be manually tested?

DataTable Sort story

#### Do the grommet docs need to be updated?

automatic

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/DataTable/DataTable.js [hunk 1]
```diff
@@ -105,9 +107,13 @@ const DataTable = ({
...
-    const ascending =
-      sort && property === sort.property ? !sort.ascending : true;
-    setSort({ property, ascending });
+    let direction;
+    if (!sort || property !== sort.property) direction = 'asc';
+    else if (sort.direction === 'asc') direction = 'desc';
+    else direction = 'asc';
+    const nextSort = { property, direction };
+    setSort(nextSort);
+    if (onSortProp) onSortProp(nextSort);
```
Comment (ShimiSun) [inline src/js/components/DataTable/DataTable.js]: I'm guessing this is an internal calculation on our end and there are no backward compatibility issues here.

File: src/js/components/DataTable/doc.js [hunk 2]
```diff
@@ -159,6 +159,13 @@ export const doc = DataTable => {
       names and values which are the search text strings. This is typically
       employed so a back-end can be used to search through the data.`,
     ),
+    onSort: PropTypes.func.description(
+      `When supplied, this function will be called with an event object
+      that includes a 'sort' property containing an object with a
+      'property' property that indicates which property
+      is being sorted on and a 'direction' property that will either be
+      'asc' or 'desc'. onChange={({ sort: { property, direction }}) => {}}`,
```
Comment (ShimiSun) [inline src/js/components/DataTable/doc.js]: why is the example not stating `onSort` to connect the context better? I think the onChange should be replaced.

File: src/js/components/DataTable/DataTable.js (hunk 1 above)
Comment (ericsoderberghp) [reply src/js/components/DataTable/DataTable.js]: correct

File: src/js/components/DataTable/doc.js (hunk 2 above)
Comment (ericsoderberghp) [reply src/js/components/DataTable/doc.js]: Good catch! I've updated the documentation.

**Comment by ericsoderberghp** [comment]
Yep, `sortable` is just there for backwards compatibility.

**Comment by ShimiSun** [comment]
Should we mark `sortable` as deprecated and provide a warning? Also, maybe just a comment on the docs that it is not the best practice and we recommend using onSort/sort instead.

**Comment by ericsoderberghp** [comment]
Hmm... Now that I think about it, if you want the current behavior, where DataTable does it all and you don't want to bother with it, you wouldn't want to set `sort` or `onSort`, to have them be functional.

**Comment by ericsoderberghp** [comment]
So, `sortable` does provide a, very minor, capability that you can't achieve with the other two. Unless, maybe we allow `sort={true}`. Seems minor.

**Comment by ShimiSun** [comment]
linking to https://github.com/grommet/grommet/issues/3728

**Pull request merged**